test/unit/models/edition_taxonomy_tag_form_test.rb

```diff
@@ -18,17 +20,45 @@ class EditionTaxonomyTagFormTest < ActiveSupport::TestCase
     form = EditionTaxonomyTagForm.load(content_id)
 
     assert_equal(form.edition_content_id, content_id)
-    assert_equal(form.taxons, taxons)
+    assert_equal(form.selected_taxons, taxons)
     assert_equal(form.previous_version, 1)
   end
 
-  test '#publish should send a patch links to publishing-api' do
+  test '#publish should send a patch links to publishing-api with the most specific selected taxons' do
```

Comment:
I wasn't clear I think. I'm suggesting something like this in `TaxonomyTagForm`:

```ruby
def taxons_to_publish
  FindTheChildest.new(education_taxons.tree,  selected_taxons).taxons
end
```

and then extract the logic into that class.

```ruby
class FindTheChildest
  def taxons
    education_taxons.each_with_object([]) do |taxon, list_of_taxons|
      content_ids = taxon.descendants.map(&:content_id)

      any_descendants_selected = selected_taxons.any? do |selected_taxon|
        content_ids.include?(selected_taxon)
      end

      unless any_descendants_selected
        content_id = taxon.content_id
        list_of_taxons << content_id if selected_taxons.include?(content_id)
      end
    end
  end
end
```

Which means that you can test this class in isolation, without relying on webmock/test helpers magic.